lib/ansiblelint/runner.py

```diff
@@ -77,29 +77,37 @@ def is_excluded(self, file_path: str) -> bool:
     def run(self) -> List[MatchError]:
         """Execute the linting process."""
         files: List[Lintable] = list()
+        matches: List[MatchError] = list()
```

Comment:
Nitpick: is there a reason why are we using a list here instead of a set? We do not want to have duplicates, so we might as well use set here and call `matches.union` instead of `matches.extend`.

Reply:
The reason why I did kept list is because I wanted to avoid changing the signature. Anyway, I am considering more refactoring in that area and maybe adopting a container that can hide the boring parts like unique, sort and maybe even retrieving stats or subsets.